### PR TITLE
fdroidserver: 1.1.9 -> 2.0.3

### PR DESCRIPTION
**Pull request by Kranzes**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129931` at e41faa06 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>fdroidserver</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'pyasn1-modules == 0.2.1' doesn't match anything in file 'setup.py'
Near pkgs/development/tools/fdroidserver/default.nix:58:0:
```
   |
58 |     description = "Server and tools for F-Droid, the Free Software repository system for Android";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/fdroidserver/default.nix [hunk 1]
```diff
@@ -4,14 +4,14 @@
...
 
   src = fetchFromGitLab {
     owner = "fdroid";
     repo = "fdroidserver";
     rev = version;
-    sha256 = "098dcg8jdi4q1prfjmd2lbhcyzb8fmmfhbxhid4kqx8vcv7r0iql";
+    sha256 = "sha256-/tX45t/DsWd0/R9VJJsqNjoOkgGIvqvq05YaVp0pLf0=";
   };
 
   patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fdroidserver/default.nix:17]:
```suggestion
  postPatch = ''
```
Also ``pyasn1-modules`` no longer matches.

File: pkgs/development/tools/fdroidserver/default.nix [hunk 2]
```diff
@@ -41,12 +41,12 @@ python.pkgs.buildPythonApplication rec {
...
-    pyasn1-modules
     python-vagrant
     pyyaml
     qrcode
     requests
     ruamel_yaml
+    yamllint
   ];
 
   # no tests
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fdroidserver/default.nix:52]:
```suggestion
  # no tests
  
  doCheck = false;
  
  pythonImportsCheck = [ "fdroidserver" ];
```

File: pkgs/development/tools/fdroidserver/default.nix [hunk 3]
```diff
@@ -41,12 +41,12 @@ python.pkgs.buildPythonApplication rec {
     paramiko
     pillow
     pyasn1
-    pyasn1-modules
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fdroidserver/default.nix:44]: This is still required https://gitlab.com/fdroid/fdroidserver/-/blob/2.0.3/setup.py#L82

File: pkgs/development/tools/fdroidserver/default.nix [hunk 4]
```diff
@@ -4,17 +4,17 @@
...
     repo = "fdroidserver";
     rev = version;
-    sha256 = "098dcg8jdi4q1prfjmd2lbhcyzb8fmmfhbxhid4kqx8vcv7r0iql";
+    sha256 = "sha256-/tX45t/DsWd0/R9VJJsqNjoOkgGIvqvq05YaVp0pLf0=";
   };
 
-  patchPhase = ''
+  postPatch = ''
     substituteInPlace fdroidserver/common.py --replace "FDROID_PATH = os.path.realpath(os.path.join(os.path.dirname(__file__), '..'))" "FDROID_PATH = '$out/bin'"
     substituteInPlace setup.py --replace "pyasn1-modules == 0.2.1" "pyasn1-modules"
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fdroidserver/default.nix:19]: This line lo longer matches this one https://gitlab.com/fdroid/fdroidserver/-/blob/2.0.3/setup.py#L82

**Pull request merged**